### PR TITLE
Return robots with type

### PR DESCRIPTION
**Pull request by janosbrodbeck**
* New DTO class `RobotWithType`
* GET Endpoints return robot with the data of its type directly
* Did not implement this for the other endpoints. Was not sure with create, if we need it, It's easy to add there as well
* Endpoint test extended to check the type as well

### CODE REVIEW

File: src/app/api/api_v1/endpoints/robot.py [hunk 1]
```diff
@@ -31,10 +33,11 @@ async def read_robot(
     """
     Retrieve robot by ID
     """
-    result = await robot.get_by_id(db=db, id=_id)
-    if not result:
-        raise HTTPException(status_code=404, detail="Robot not found")
-    return result
+    robot_obj = await robot.get_by_id(db=db, id=_id)
+    if not robot_obj:
+        raise HTTPException(status_code=404, detail="Robo1t not found")
```
Comment (dolevi-tub) [inline src/app/api/api_v1/endpoints/robot.py:38]:
```suggestion
        raise HTTPException(status_code=404, detail="Robot not found")
```
Typo

**Review by dolevi-tub — CHANGES_REQUESTED** [review]
Looks good apart frmo the typo :+1:

File: src/app/api/api_v1/endpoints/robot.py (hunk 1 above)
Comment (janosbrodbeck) [reply src/app/api/api_v1/endpoints/robot.py]: Fixed

**Review by dolevi-tub — APPROVED** [review]
Good now!

**Pull request merged**